### PR TITLE
feat: handle active subs in other sites when cancelling memberships

### PR DESCRIPTION
**Pull request by leogermani**
Follow instructions at https://github.com/Automattic/newspack-network/pull/122

### CODE REVIEW

File: tests/unit-tests/test-subscription-integration.php [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+		$this->assertEquals( [], $result );
+
+		$result = $method->invoke( null, self::$user_with_non_network_sub_on_1 );
+		$this->assertCount( 1, $result );
+		$this->assertArrayHasKey( 'http://site1', $result );
+		$this->assertCount( 1, $result['http://site1'] );
+		$this->assertArrayHasKey( 503, $result['http://site1'] );
+
+		$result = $method->invoke( null, self::$user_without_subs );
+		$this->assertEquals( [], $result );
```
Comment (adekbadek) [inline tests/unit-tests/test-subscription-integration.php]: These tests are a bit hard to read with the generic variable names. It'd be more readable if `$method` was called e.g. `$get_user_network_active_subscriptions_method` and `$result` e.g. `$user_network_active_subscriptions`.

**Review by adekbadek — APPROVED** [review]
Works as expected, but the `Membership is not cancelled…` note was appended three times to the membership, which seems excessive :)

image

File: tests/unit-tests/test-subscription-integration.php (hunk 1 above)
Comment (leogermani) [reply tests/unit-tests/test-subscription-integration.php]: https://github.com/Automattic/newspack-network/pull/125/commits/682f694368e9f1471de184e06fdbf822bd6b1a07

**Comment by leogermani** [comment]
> Works as expected, but the `Membership is not cancelled…` note was appended three times to the membership, which seems excessive :)
> 
> image

Could not replicate this in my tests...

**Comment by adekbadek** [comment]
When I tested it now, the note was added twice, instead of thrice ¯\_(ツ)_/¯. Perhaps because of the two filters used with the `prevent_membership_expiration` callback? Anyway, this is not that big of an issue, and mitigation might add unnecessary complexity.

**Review by adekbadek — APPROVED** [review]

**Comment by leogermani** [comment]
Closing in favor of https://github.com/Automattic/newspack-network/pull/135

**Pull request closed without merging**